probe-rs/src/architecture/arm/traits.rs

```diff
@@ -45,21 +45,66 @@ pub enum DpAddress {
     Multidrop(u32),
 }
 
+/// Access port v2 address
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum ApV2Address {
+    /// Last node of an APv2 address
+    Leaf(u32),
+    /// Non-terminal node of an APv2 address
+    Node(u32, Box<ApV2Address>),
```

Comment:
Could this maybe be represented as a `vec`? This seems a bit involved if we only have chains of addresses.

Reply:
Having nested MemAP is not recommended by the specification (see the quote [here](https://github.com/probe-rs/probe-rs/pull/2600#issuecomment-2198633032)), so I would’d expect the nested case to be the most common occurence.

I haven’t bench marked if there’s any difference between:
```rust
match apv2 {
	ApV2Address::Leaf(u) => …,
	ApV2Address::Node(v, r) => …,
}
```
and
```rust
match apv2 {
	[ u ] => …,
	[ u, r @ .. ] => …,
}
```

but IMHO the former reads better (without extra knowledge/context) than the latter option.